### PR TITLE
Improve TagContextImpl tests.

### PR DESCRIPTION
**Pull request by sebright**
This commit splits some of the existing tests in order to test different
operations separately. It also updates variable names and the implementation of
TagContextBuilder.put after the removal of tag types.

### CODE REVIEW

File: impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java [hunk 1]
```diff
@@ -38,55 +37,81 @@
...
+  }
+
+  @Test
+  public void getTags_nonEmpty() {
+    TagContextImpl tags = new TagContextImpl(ImmutableMap.of(K1, V1, K2, V2));
+    assertThat(tags.getTags()).containsExactly(K1, V1, K2, V2);
+  }
+
+  @Test
+  public void put_newKey() {
```
Comment (songy23) [inline impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: Add tests on putting null key and null value?

File: impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java [hunk 2]
```diff
@@ -38,55 +37,81 @@
...
+        .containsExactly(K2, V2);
   }
 
   @Test
-  public void testClear() {
-    TagContext tags = tagger.emptyBuilder().put(KS1, V1).build();
-    assertThat(tagContextToList(tagger.toBuilder(tags).remove(KS1).build())).isEmpty();
-    assertThat(tagContextToList(tagger.toBuilder(tags).remove(KS2).build()))
-        .containsExactly(Tag.create(KS1, V1));
+  public void remove_differentKey() {
```
Comment (songy23) [inline impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: Similar, add tests on removing null?

File: impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java (hunk 1 above)
Comment (sebright) [reply impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: Good catch!

File: impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java (hunk 2 above)
Comment (sebright) [reply impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: Done

**Comment by sebright** [comment]
Thanks!

File: impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java [hunk 3]
```diff
@@ -38,55 +38,108 @@
...
+        .containsExactly(K1, V1);
   }
 
   @Test
-  public void testClear() {
-    TagContext tags = tagger.emptyBuilder().put(KS1, V1).build();
-    assertThat(tagContextToList(tagger.toBuilder(tags).remove(KS1).build())).isEmpty();
-    assertThat(tagContextToList(tagger.toBuilder(tags).remove(KS2).build()))
-        .containsExactly(Tag.create(KS1, V1));
+  public void remove_nullValue() {
```
Comment (HailongWen) [inline impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: should `remove_nullKey` be better since we are testing against the key?

Comment (sebright) [reply impl_core/src/test/java/io/opencensus/implcore/tags/TagContextImplTest.java]: Thanks, I made a PR: #910
